fmt/printf.h

```diff
@@ -588,6 +600,8 @@ FMT_VARIADIC(int, printf, CStringRef)
   \endrst
  */
 inline int fprintf(std::ostream &os, CStringRef format_str, ArgList args) {
+  std::mutex mtx;
+  std::unique_lock<std::mutex> lck(mtx);
```

Comment:
Sure but if you want to synchronize access to `std::ostream` then the mutex should be placed elsewhere. If you put it in this function, you'll create a new one on every call. Moreover, I think it belongs in the client code and not in the library.